contracts/strategy/RWAStrategy.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright Fathom 2024
+pragma solidity 0.8.19;
+
+import "./BaseStrategy.sol";
+import { ERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+
+// solhint-disable
+contract RWAStrategy is BaseStrategy {
+    using SafeERC20 for ERC20;
+    using Math for uint256;
+
+    // Mapping to be set by management for any reward tokens.
+    // This can be used to set different mins for different tokens
+    // or to set to uint256.max if selling a reward token is reverting
+    // to allow for reports to still work properly.
+    uint256 public minAmountToSell;
+
+    // The address that we send funds to
+    // and that will buy RWAs with the funds
+    address public immutable managerAddress;
+
+    uint256 public totalInvestedInRWA;
+
+    constructor(address _asset, string memory _name, address _tokenizedStrategyAddress, address _managerAddress, uint256 _minAmountToSell) BaseStrategy(_asset, _name, _tokenizedStrategyAddress) {
+        managerAddress = _managerAddress;
+        minAmountToSell = _minAmountToSell;
+    }
+
+    function _harvestAndReport() internal override returns (uint256 _totalAssets) {
```

Comment:
So harvestAndReport is taking assets to the manager? But where is the place where the gain/loss is generated?

Reply:
Every time manage has a gain he will simply deposit the gains back on the RWA strategy. Need to think about a loss, don't have an answer for that now.